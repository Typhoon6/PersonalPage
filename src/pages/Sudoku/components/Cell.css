
.cell {
   width: 50px;
   height: 50px;
   font-size: 26px;
   text-align: center;
   border: 1px solid black;
}

.cell-highlighted {
    width: 50px;
    height: 50px;
    font-size: 26px;
    text-align: center;
    border: 1px solid black;
    background-color: #dedede;
}