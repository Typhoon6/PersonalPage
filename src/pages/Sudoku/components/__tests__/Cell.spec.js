import React from 'react';
import Cell from '../Cell';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('<Cell />', () => {
    it('renders empty', () => {
        const component = renderer.create(
            <Cell />
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('renders highlighted', () => {
        const component = renderer.create(
            <Cell highlighted={true}/>
        )
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('calls action after click', () => {
        let selectCellWasCalled = false
        let component = shallow(
          <Cell selectCell={() => {
              selectCellWasCalled = true
              }}/>
        );
      
        component.find('input').simulate('focus');
        expect(selectCellWasCalled).toBe(true);
    });

    it('calls action after value change with number', () => {
        let setValueWasCalled = false
        let component = shallow(
          <Cell setValue={
              () => {
                setValueWasCalled = true
              }
          }/>
        );
      
        component.find('input').simulate('change', {
            target: {
                value: '9'
            }
        });
        expect(setValueWasCalled).toBe(true);
    });
    it('does not call action after value change with non number', () => {
        let setValueWasCalled = false
        let component = shallow(
          <Cell setValue={
              () => {
                setValueWasCalled = true
              }
          }/>
        );
      
        component.find('input').simulate('change', {
            target: {
                value: 'p'
            }
        });
        expect(setValueWasCalled).toBe(false);
    });
})