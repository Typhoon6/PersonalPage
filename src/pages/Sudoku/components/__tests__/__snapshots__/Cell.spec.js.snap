// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> renders empty 1`] = `
<input
  className="cell"
  disabled={undefined}
  onChange={[Function]}
  onFocus={[Function]}
  value={undefined}
/>
`;

exports[`<Cell /> renders highlighted 1`] = `
<input
  className="cell-highlighted"
  disabled={undefined}
  onChange={[Function]}
  onFocus={[Function]}
  value={undefined}
/>
`;
